### ISSUE TITLE
Unknown utility "bg-sky-800" in Twind

### ISSUE THREAD

**Issue by gmunumel** [denoland/fresh#801]
Hello, 

I upgraded to version 1.1 and start getting this error (was not present in last version): 

![Screenshot 2022-09-26 185254](https://user-images.githubusercontent.com/276168/192335692-a5e24af2-a2d8-4537-ac79-055d64f242f4.png)

my **twind.config.ts**:

```
import { Options } from "$fresh/plugins/twind.ts";
import * as colors from "twind/colors";

export default {
  selfURL: import.meta.url,
  darkMode: "class",
  mode: "silent",
  theme: {
    extend: {
      colors,
    },
  },
} as Options;
```

```
deno 1.25.0 (release, x86_64-pc-windows-msvc)
v8 10.6.194.5
typescript 4.7.4
```

**Comment by mooxl** [comment]
<details><summary>This happens to all custom theme values.</summary>
<img width="997" alt="Screenshot 2022-09-27 at 10 20 17" src="https://user-images.githubusercontent.com/35741000/192473468-5c4688c1-fb0c-490e-853a-6efdd027a5b1.png">
</details>

<details><summary>It seems that the default theme is always used for linting and intellisense, even though it has been overwritten.</summary>
<img width="997" alt="Screenshot 2022-09-27 at 10 20 46" src="https://user-images.githubusercontent.com/35741000/192473500-c0c5ec85-2b2f-4a0c-8e1e-6f91b911bed5.png">
</details>


```
deno 1.25.4 (release, aarch64-apple-darwin)
v8 10.7.193.3
typescript 4.7.4
```

**Comment by victorkane** [comment]
I can verify that when I migrated the @lucacasonato You Tube video blog code (gh lucacasonato/freshworkshop) to Fresh 1.1 (twind syntax, other changes reflected in Deno blog post announcing Fresh 1.1) I got the same problem redlined in reference to ¨markdown-body¨; and that same code can be seen in the recently showcased blog template https://github.com/littlesticks/fresh-blog (which uses Fresh 1.1) in `routes/blog/[slug].md` invoking same ¨markdown-body¨.

**Comment by mooxl** [comment]
It turned out that the plugin works perfectly. It is most likely due to the vscode twind extension: https://github.com/tw-in-js/vscode-twind-intellisense/issues/18

**Comment by kitsonk** [comment]
Yes, the comment here gives the workaround: https://github.com/tw-in-js/vscode-twind-intellisense/issues/18#issuecomment-1264023267 but it isn't totally clear in the details.

On the latest version of fresh what you need to do is:

- remove `selfUrl` from `/twind.config.ts`
- remove the import of `Options` from `$fresh/plugins/twind.ts`.
- apply a type to the default export via JSDoc like this:

  ```ts
  /** @type {Omit<import("$fresh/plugins/twind.ts").Options, "selfURL">} */
  ```

- Modify the `start` arguments in `main.ts` to look like this:

  ```ts
  await start(manifest, {
    plugins: [twindPlugin({
      selfURL: new URL("./twind.config.ts", import.meta.url).href,
      ...twindConfig,
    })],
  });
  ```

Then configuration changes to the `/twind.config.ts` should be reflected in the intellisense in the plugin (at least they have for me).

What appears to be going on is that when the fresh twind gets pulled in, it is "overriding" the inferred types that the `vscode-twind-intellisense` gets from the twind config, meaning only "built in" values are checked.  By excluding the hard import and using the JSDoc "type import" it seems to free up the plugin to properly infer the config.

I am not totally sure how easy it would be to make this the default @lucacasonato.  I might take a look later, but it does seem to be a bit of a friction point and seems to be a common pattern to adjust your twind config.

**Comment by pheuter** [comment]
To get plugin intellisense working with the expanded tailwindcss color palette, I had to modify the `import_map.json` file to use the `npm:` package specifier as well as pass `--node-modules-dir` to the task:

**import_map.json**:

```json
{
  "imports": {
    "$fresh/": "https://deno.land/x/fresh@1.1.2/",
    "preact": "https://esm.sh/preact@10.11.0",
    "preact/": "https://esm.sh/preact@10.11.0/",
    "preact-render-to-string": "https://esm.sh/*preact-render-to-string@5.2.4",
    "@preact/signals": "https://esm.sh/*@preact/signals@1.0.3",
    "@preact/signals-core": "https://esm.sh/*@preact/signals-core@1.0.1",
    "twind": "npm:twind@0.16.17",
    "twind/": "https://esm.sh/twind@0.16.17/"
  }
}
```

**deno.json:**

```json
{
  "tasks": {
    "start": "deno run --unstable --node-modules-dir -A --watch=static/,routes/ dev.ts"
  },
  "importMap": "./import_map.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "preact"
  }
}
```

**twind.config.ts:**

```ts
import * as colors from "twind/colors";

export default {
  theme: {
    extend: {
      colors,
    },
  },
};
```

That way, the vscode plugin is able to locate `twind/colors` inside node_modules for intellisense and Deno/Fresh is still able to properly init the twind fresh plugin.

**Comment by tristanisham** [comment]
To clarify the previous solution. You can just replace your tailwind.config.ts with this and all you'll have to suffer are some red-lines.
```ts
import { Options } from "$fresh/plugins/twind.ts";
import * as colors from "twind/colors";


export default {
  selfURL: import.meta.url,
  theme: {
    extend: {
      colors,
    },
  },
} as Options;
```

**Comment by mcdaded** [comment]
> Yes, the comment here gives the workaround: [tw-in-js/vscode-twind-intellisense#18 (comment)](https://github.com/tw-in-js/vscode-twind-intellisense/issues/18#issuecomment-1264023267) but it isn't totally clear in the details.
> 
> On the latest version of fresh what you need to do is:
> 
> * remove `selfUrl` from `/twind.config.ts`
> * remove the import of `Options` from `$fresh/plugins/twind.ts`.
> * apply a type to the default export via JSDoc like this:
>   ```ts
>   /** @type {Omit<import("$fresh/plugins/twind.ts").Options, "selfURL">} */
>   ```
> * Modify the `start` arguments in `main.ts` to look like this:
>   ```ts
>   await start(manifest, {
>     plugins: [twindPlugin({
>       selfURL: new URL("./twind.config.ts", import.meta.url).href,
>       ...twindConfig,
>     })],
>   });
>   ```
> 
> Then configuration changes to the `/twind.config.ts` should be reflected in the intellisense in the plugin (at least they have for me).
> 
> What appears to be going on is that when the fresh twind gets pulled in, it is "overriding" the inferred types that the `vscode-twind-intellisense` gets from the twind config, meaning only "built in" values are checked. By excluding the hard import and using the JSDoc "type import" it seems to free up the plugin to properly infer the config.
> 
> I am not totally sure how easy it would be to make this the default @lucacasonato. I might take a look later, but it does seem to be a bit of a friction point and seems to be a common pattern to adjust your twind config.

This solved my issue thanks!

**Comment by marvinhagemeister** [comment]
Closing this as we've switched to Tailwind CSS which allows us to compile the style sheet ahead of time and move it out of the render path. The Tailwind CSS plugin will be released in the next Fresh 1.6 version which I'm hoping to release later this week.
